tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -123,8 +124,9 @@ public static void Initialize()
             {
                 Log.Debug("Sending CallTarget derived integration definitions to native library.");
                 var payload = InstrumentationDefinitions.GetDerivedDefinitions();
-                NativeMethods.AddDerivedInstrumentations(payload.DefinitionsId, payload.Definitions);
-                foreach (var def in payload.Definitions)
+                var definitionsArray = GetDefinitions(payload);
```

Comment:
I would have thought we would initialized only the `NoFilter` methods.
Then upon AppSec Initialization, we would have initialized the native methods needed for appsec only. This would have avoided to wait for appsec init and this is what is done for Serverless.

But looking at your review, I wonder if what you're doing is better because:
- an instrumentation could be used by several all at some point. Meaning we shall handle that use case in the native code. (though we don't have a use case where CiApp and AppSec overlap so far). 
- I don't know if we care about waiting for appsec init actually.

Do you have any opinion on this? If not we can ask the second reviewer what he/she thinks.

(Note that if we go for this implem, we shall refactor a bit cause calling the AppSec Init after is useless now, and also we were doing the init from a try/catch block and we don't do it here).

Reply:
good idea, that would be better indeed if we could just initialize the `NoFilter`, and if Appsec loaded right, then rejit appsec instrumentations. Maybe I could call a `NativeMethods.AddAppSecInstrumentations()` from the Security constructor once we know for sure appsec is enabled (that the waf was found etc, not just the env variable), wdyt? maybe @tonyredondo  would have some preference?

Reply:
I believe that as long as we don't have overlap between instrumentations (and we don't have a use case for that now) then it would be better to do it the Serverless way indeed

Reply:
changed in [9190cfa](https://github.com/DataDog/dd-trace-dotnet/pull/2580/commits/9190cfa46929f5876bf5525680e1e88ad6df6d67) , let met know wyt :)